hub.py

```diff
@@ -325,6 +351,27 @@ def deploy(self, auth_provider, proxy_secret_key):
                 '-f', generated_values_file.name,
                 '-f', values_file.name,
             ]
+
             print(f"Running {' '.join(cmd)}")
             subprocess.check_call(cmd)
 
+            if test_notebook:
+                try_idx = 1
+                if self.spec['template'] != 'base-hub':
+                    service_api_token = generated_values["base-hub"]["jupyterhub"]["hub"]["services"]["hub-health"]["apiToken"]
+                else:
+                    service_api_token = generated_values["jupyterhub"]["hub"]["services"]["hub-health"]["apiToken"]
+                # Try 2 times before declaring it a failure
+                while try_idx <= 2:
+                    try:
+                        print(f"Validate hub health try {try_idx} started...")
```

Comment:
This is ok for now, but let's add a FIXME to move to something like https://github.com/litl/backoff? Doing retries properly is complicated and hard, we should just re-use other people's work :D